src/seedu/addressbook/AddressBook.java

```diff
@@ -431,6 +434,20 @@ private static String executeAddPerson(String commandArgs) {
         addPersonToAddressBook(personToAdd);
         return getMessageForSuccessfulAddPerson(personToAdd);
     }
+	
+	private static boolean executeAddPersonAfterEdit(String commandArgs, int index) {
+        // try decoding a person from the raw args
+        final Optional<HashMap<String, String>> decodeResult = decodePersonFromString(commandArgs);
+        // checks if args are valid (decode result will not be present if the person is invalid)
+        if (!decodeResult.isPresent()) {
```

Comment:
When you modify the behavior, remember to update documentation and tests as well.